src/shackle_pool.erl

```diff
@@ -84,26 +84,57 @@ init() ->
 
 server(Name) ->
     case options(Name) of
-        {ok, #pool_options {
-                backlog_size = BacklogSize,
-                client = Client,
-                pool_size = PoolSize,
-                pool_strategy = PoolStrategy
-            }} ->
-
-            ServerIndex = server_index(Name, PoolSize, PoolStrategy),
-            Server = shackle_pool_utils:server_name(Name, ServerIndex),
-            case shackle_backlog:check(Server, BacklogSize) of
-                true ->
-                    {ok, Client, Server};
-                false ->
-                    {error, backlog_full}
-            end;
-        {error, Reson} ->
-            {error, Reson}
+        {ok, Options} ->
+            choose_server(Name, Options);
+        {error, Reason} ->
+            {error, Reason}
     end.
 
 %% private
+choose_server(Name,
+              #pool_options {
+                 backlog_size = BacklogSize,
+                 client = Client,
+                 pool_size = PoolSize,
+                 pool_strategy = two_choice
+                }) ->
+    {IndexA, IndexB} =
+        {server_index(Name, PoolSize, random),
+         server_index(Name, PoolSize, random)},
+    {ServerA, ServerB} =
+        {shackle_pool_utils:server_name(Name, IndexA),
+         shackle_pool_utils:server_name(Name, IndexB)},
+    Backlogs =
+        {shackle_backlog:check(ServerA, BacklogSize),
+         shackle_backlog:check(ServerB, BacklogSize)},
```

Comment:
By doing a check, you're basically "reserving" a spot on the worker. So calling it twice means you might actually return `{error, backlog_full}` more often.